libraries/core_libs/consensus/src/final_chain/rewards_stats.cpp

```diff
@@ -66,7 +64,9 @@ void RewardsStats::initStats(const PeriodData& sync_blk) {
     }
   }
 
-  // TODO: add votes to rewards_stats
+  for (const auto& vote : sync_blk.previous_block_cert_votes) {
+    addVote(vote);
```

Comment:
Why do we need to have bonus_count info inside PeriodData ? Isnt it enough to calculate it here and set it to rewards_stats as this info i only needed in dpos contract ?

Reply:
you do not have vote weigh in this time as it is calculated on vote verification. e.g. after restarting node some pbft blocks could be finalized but not executed, so you would need to revalidate all the votes again

Reply:
So during normal node operation, these votes are already verified and for the " node restart scenario", isnt the max possible difference between finalized & executed blocks just a few blocks ? If yes, I think it is worth to do this verification during node init as it is 1 time operation and it shouldnt really take too much time

Reply:
After https://github.com/Taraxa-project/taraxa-node/pull/1818 merged  I will change this architecture